micrometer-core/src/main/java/io/micrometer/core/instrument/distribution/DistributionStatisticConfig.java

```diff
@@ -110,7 +117,11 @@ public NavigableSet<Double> getHistogramBuckets(boolean supportsAggregablePercen
         NavigableSet<Double> buckets = new TreeSet<>();
 
         if (percentileHistogram != null && percentileHistogram && supportsAggregablePercentiles) {
-            buckets.addAll(PercentileHistogramBuckets.buckets(this));
+            long scale = baseTimeUnit.toMillis(1);
```

Comment:
Trying to understand why this is a problem? I believe this is already taken care of since the buckets are always converted to Nanos and recording happens in Nanos. When exporting things are again converted to the baseTimeunit(). Am I missing something?

Reply:
I just double checked and saw that I'm using the a different histogram implementation that I thought I had (the one from the OpenTelemetry SDK).